source/controllers/include/controllers/IController.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <list>
+#include <map>
+#include <memory>
+
+#include "controllers/exceptions/NoRobotModelException.hpp"
+
+#include "robot_model/Model.hpp"
+#include "state_representation/parameters/ParameterMap.hpp"
+#include "state_representation/space/Jacobian.hpp"
+#include "state_representation/space/joint/JointPositions.hpp"
+#include "state_representation/space/joint/JointState.hpp"
+
+/**
+ * @namespace controllers
+ * @brief Systems to determine a command output from measured and desired inputs
+ */
+namespace controllers {
+/**
+ * @class IController
+ * @brief Abstract class to define a controller in a desired state type, such as joint or Cartesian spaces
+ * @tparam S the state type of the controller
+ */
+template<class S>
+class IController : public state_representation::ParameterMap {
+public:
+  /**
+   * @brief Empty constructor
+   */
+  IController() = default;
```

Comment:
You might need to add a constructor with a robot model if we want to remove the robot model setter (see my other comment).